File: Models/Vehicle.swift
//
//  Vehicle.swift
//  UDrive
//
//  Created on 2024
//

import Foundation

struct Vehicle: Identifiable, Codable {
    let id: UUID
    var make: String
    var model: String
    var year: Int
    var color: String
    var licensePlate: String
    var capacity: Int
    var vehicleImageURL: String?
    
    init(id: UUID = UUID(), make: String, model: String, year: Int, color: String, licensePlate: String, capacity: Int, vehicleImageURL: String? = nil) {
        self.id = id
        self.make = make
        self.model = model
        self.year = year
        self.color = color
        self.licensePlate = licensePlate
        self.capacity = capacity
        self.vehicleImageURL = vehicleImageURL
    }
}

File: lib/models/vehicle.dart
class Vehicle {
  final String id;
  final String make;
  final String model;
  final int year;
  final String color;
  final String licensePlate;
  final int capacity;
  final String? vehicleImageUrl;

  Vehicle({
    required this.id,
    required this.make,
    required this.model,
    required this.year,
    required this.color,
    required this.licensePlate,
    required this.capacity,
    this.vehicleImageUrl,
  });

  factory Vehicle.fromJson(Map<String, dynamic> json) {
    return Vehicle(
      id: json['id'] as String,
      make: json['make'] as String,
      model: json['model'] as String,
      year: json['year'] as int,
      color: json['color'] as String,
      licensePlate: json['licensePlate'] as String,
      capacity: json['capacity'] as int,
      vehicleImageUrl: json['vehicleImageUrl'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'make': make,
      'model': model,
      'year': year,
      'color': color,
      'licensePlate': licensePlate,
      'capacity': capacity,
      'vehicleImageUrl': vehicleImageUrl,
    };
  }
}

